app/src/main/java/com/github/fribourgsdp/radio/data/view/UserProfileActivity.kt

```diff
@@ -122,9 +122,11 @@ open class UserProfileActivity : MyAppCompatActivity(), KeepOrDismissPlaylistDia
         if(user.isGoogleUser){
             db.setUser(user.id,user)
         }else{
+            val userPlaylists = user.getPlaylists()
             val userWithoutPlaylist = user
-            userWithoutPlaylist.removePlaylists(user.getPlaylists())
+            userWithoutPlaylist.removePlaylists(userPlaylists)
             db.setUser(user.id,userWithoutPlaylist)
+            user.addPlaylists(userPlaylists)
```

Comment:
Why remove the playlist to then add them later?

Reply:
When an user is not signed in, only his username is saved online, not his playlists, this is why I need to remove all the  playlists from the user before saving it. I have to reput the playlist in the original user as in reality userWithoutPlaylist and  user refer to the same entity. I create the UserWithoutPlaylist just for clarity reason.